DietPi-Software | Add Homer dashboard

MichaIng opened the pull request.

t3dium committed:
new package - mkcert

t3dium committed:
Merge branch 'MichaIng:master' into master

t3dium committed:
new package - mkcert

t3dium committed:
[dietpi software] homer, nginx proxy manager & psad

t3dium committed:
fixed github messing with my indents....

t3dium committed:
a minor bug fix

sudo docker start $(docker ps -a -q) - >  sudo docker start $(sudo docker ps -a -q), missed the 2nd sudo so it didn't previously work

t3dium committed:
a minor bug fix

t3dium committed:
homer

t3dium committed:
Merge branch 'dev' into master

t3dium committed:
Update dietpi-software

t3dium committed:
port 80, subdirectory so ip/homer

t3dium committed:
Update dietpi-software

t3dium committed:
backup path change

t3dium committed:
bug

homer - copying to nonexistent directory

Joulinar committed:
Merge branch 'master' of https://github.com/t3dium/DietPi into pr/5827

MichaIng committed:
v8.11

- DietPi-Software | Homer: Apply suggestions from #5829

MichaIng committed:
v8.11

- DietPi-Software | Homer: Remove old assets with hash in file name on reinstalls, to not accumulate obsolete files

MichaIng committed:
v8.11

- DietPi-Software | Homer: Assure as well that obsolete assets with hash in file name do not accumulate in Homer backup

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -12318,6 +12325,27 @@ _EOF_
+				[[ -d '/mnt/dietpi_userdata/homer_backup/resources' ]] && G_EXEC rm -Rf /mnt/dietpi_userdata/homer_backup/{workbox-*.js,resources}
+				G_EXEC cp -a /var/www/homer/. /mnt/dietpi_userdata/homer_backup/
+				G_DIETPI-NOTIFY 2 'Removing old assets with hash in file name ...'
+				G_EXEC rm -Rf /var/www/homer/{workbox-*.js,resources}
```
Why not removing the old assets before creating the backup? This way we would need to remove them only once?

MichaIng replied:
Right, but then the backup wouldn't contain any of those assets. It is mainly to restore those assets/configs users might edit, but its another bonus that the backup is a complete one, just in case the download fails.

Better would be actually to download to tmpfs first, and do all the backup creation and install afterwards, so in case of failing download the untouched old instance is still in place 🤔.

Joulinar replied:
hmm but we do the same removal on backup as well as on live folder, isn't it`?

```
rm -Rf /mnt/dietpi_userdata/homer_backup/{workbox-*.js,resources}
rm -Rf /var/www/homer/{workbox-*.js,resources}
```

means we could remove them before already? Or do I'm missing something?

MichaIng replied:
> hmm but we do the same removal on backup as well as on live folder, isn't it`?

Jep, but the assets from the active instance are then backed up, together with the new `index.html` which loads these assets.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -12318,6 +12325,27 @@ _EOF_
+			fi
+		
+			# Install latest release, preserve existing config
+			Download_Install 'https://github.com/bastienwirtz/homer/releases/latest/download/homer.zip' /var/www/homer
```
```suggestion
			# Download the latest release
			Download_Install 'https://github.com/bastienwirtz/homer/releases/latest/download/homer.zip' homer

			# Backup existing instance
			if [[ -d '/var/www/homer' ]]
			then
				G_DIETPI-NOTIFY 2 'Existing Homer instance found, backing up to /mnt/dietpi_userdata/homer_backup ...'
				[[ -d '/mnt/dietpi_userdata/homer_backup' ]] || G_EXEC rm -R /mnt/dietpi_userdata/homer_backup
				G_EXEC cp -a /var/www/homer /mnt/dietpi_userdata/homer_backup
				G_DIETPI-NOTIFY 2 'Removing old assets with hash in file name ...'
				G_EXEC rm -Rf /var/www/homer/{workbox-*.js,resources}
			fi
		
			# Install latest release, preserve existing config
			G_EXEC cp -a homer/. /var/www/homer/
			G_EXEC rm -R homer
```

MichaIng replied:
@Joulinar 
Here the alternative suggested above.

Joulinar replied:
ah yeah that seems to be more logic 👍

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -15140,6 +15168,13 @@ _EOF_
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == -1 ))
+		then	
+			Banner_Uninstalling
+			[[ -d '/var/www/homer' ]] && G_EXEC rm -R /var/www/homer
```
Shall we remove the backup as well on uninstall?

Joulinar replied:
good question. Usually, I would say yes because we don't keep any backup for one of our apps. Except Own-/Nextcloud.

t3dium replied:
that'd be good yeah

MichaIng replied:
We also do not create backups in any other case, AFAIK. Ah, we can ask the user and print an info otherwise (also in case of non-interactive uninstall.

Joulinar replied:
let's keep it simple. If someone decide to uninstall, he needs to be clear everything will be purged.

MichaIng replied:
Too late 😄. It's fine, not too much code.

Joulinar replied:
ok let's do it that way.

MichaIng committed:
v8.11

- DietPi-Dashboard | Homer: Update backup and touch existing instance on reinstalls after the download has succeeded, so if download fails, the existing instance remains untouched and functional

MichaIng committed:
v8.11

- DietPi-Software | Homer: Offer to remove the Homer backup on uninstall

t3dium commented on dietpi/dietpi-software:
```diff
@@ -970,6 +970,13 @@ Available commands:
 		#------------------
+		software_id=206
+		aSOFTWARE_NAME[$software_id]='Homer'
+		aSOFTWARE_DESC[$software_id]='Simple HOMepage for your servER to keep your services on hand'
```
homepage and server shouldn't be capitalised here, rest looks good

MichaIng replied:
I took this from their GitHub repo, it explains the anagram "HOMER" from "HOMepage" and "servER" 🙂.

t3dium replied:
ohh, didn't see that lol

MichaIng committed:
v8.11

- DietPi-Software | Homer: Add changelog, survey report support, README entry and reduce software ID to 205
